### PR TITLE
first draft

### PR DESCRIPTION
**Pull request by eea-edo**

### CODE REVIEW

File: src/main/java/org/opendatanode/plugins/transformer/exceltocsv/ExcelToCsv.java [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+        CSVWriter csvWriter = null;
+        try {
+            csvWriter = new CSVWriter(new OutputStreamWriter(new FileOutputStream(csvFile), "utf-8"));
+
+            int minCellNumber = getMinCellNumber(sheet);
+            for (int r = sheet.getFirstRowNum(); r <= sheet.getLastRowNum(); r++) {
+                Row row = sheet.getRow(r);
+                if (row != null) {
+                    List<String> csvLine = new ArrayList<>();
+                    for (int c = minCellNumber; c < row.getLastCellNum(); c++) {
```
Comment (Jan-Marcek) [inline src/main/java/org/opendatanode/plugins/transformer/exceltocsv/ExcelToCsv.java:149]: Don't need to loop over all items in list. When we find first non-empty cell then we can stop.

File: src/main/java/org/opendatanode/plugins/transformer/exceltocsv/ExcelToCsv.java [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+     * @param fileName File name, eg. example.xlsx
+     * @return Base file name (file name without extension).
+     */
+    private String getBaseFileName(String fileName) {
+        // for "example.xlsx" parts is ["example", "xlsx"]
+        String[] parts = fileName.split("\\.(?=[^\\.]+$)");
+        return parts[0];
+    }
+
+    private void addFileToOutput(DpuFile file) throws DataUnitException {
```
Comment (Jan-Marcek) [inline src/main/java/org/opendatanode/plugins/transformer/exceltocsv/ExcelToCsv.java]: an unused method.

File: src/main/java/org/opendatanode/plugins/transformer/exceltocsv/ExcelToCsv.java [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+                } else {
+                    ContextUtils.sendShortInfo(ctx, "ExcelToCsv.sheetIgnored", sheet.getSheetName());
+                }
+            }
+        } catch (EncryptedDocumentException | InvalidFormatException | IOException ex) {
+            ContextUtils.sendError(ctx, "ExcelToCsv.excelTransformationFailed", ex.toString(), entry.getSymbolicName());
+            throw ex;
+        } finally {
+            if (wb != null) {
+                wb.close();
```
Comment (Jan-Marcek) [inline src/main/java/org/opendatanode/plugins/transformer/exceltocsv/ExcelToCsv.java]: IntelliJ says "IOException' might be thrown inside 'finally' "

File: src/main/java/org/opendatanode/plugins/transformer/exceltocsv/ExcelToCsv.java [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+                        Cell cell = row.getCell(c);
+                        String value = getCellValue(cell);
+                        csvLine.add(value);
+                    }
+                    csvWriter.writeNext(csvLine.toArray(empty_array));
+                }
+            }
+        } finally {
+            if (csvWriter != null) {
+                csvWriter.close();
```
Comment (Jan-Marcek) [inline src/main/java/org/opendatanode/plugins/transformer/exceltocsv/ExcelToCsv.java]: IntelliJ says "IOException' might be thrown inside 'finally' "

**Comment by hanecak** [comment]
Depends on https://github.com/UnifiedViews/Plugins/pull/389 and https://github.com/UnifiedViews/Core/pull/594 .

**Comment by hanecak** [comment]
I've moved the DPU directly into UnifiedViews/Plugins repository, see https://github.com/UnifiedViews/Plugins/pull/391 .

Thus this particular pull request is no longer needed.

Later, I will also delete this repository.

**Pull request closed without merging**